client/src/components/Visualization/charts/ScatterPlot.js

```diff
@@ -86,7 +86,7 @@ function useAdvertisers(year, queryLimit) {
 
         setAdvertisers(newAdvertisers);
       });
-  }, [advertisers, year, queryLimit]);
```

Comment:
So what exactly is the variable used for?

Reply:
tl;dr: `advertisers` is a state variable tracking the current state of `advertisers` data loaded in from the API, and is set by `setAdvertisers` in `useEffect` . This variable is then returned by `useAdvertisers`.  

[react state hook](https://reactjs.org/docs/hooks-state.html)

Reply:
I get it now. useState gives [currentState, functionToOverwriteTheCurrentState]